7a_temp_coop_munge/src/data_parsers/parse_mo_usace.R

```diff
@@ -0,0 +1,427 @@
+  }
+
+  # remove sheets with no data
+  dat <- dat[sapply(dat, function(x) dim(x)[1]) > 0]
+
+  # some sheet names have extra whitespace and inconsistent case usage
+  names(dat) <- trimws(names(dat)) %>% toupper()
+
+  return(dat)
+}
+
+#' Convert sheet column names to lower case
+#'
+#' Column names use inconsistent case within and across workbooks.
+#' This function resolves this issue.
+#'
+#' @param ls_df a list of `data.frames` generated by reading in excel sheets
+#'
+sheet_cols_to_lower <- function(ls_df) {
+  lapply(ls_df, function(x){
+    names(x) <- tolower(names(x))
+    return(x)
+  })
+}
+
+#' Remove manually identified sheets that are not necessary
+#'
+#' Several workbooks have superflous sheets that need to be removed.
+#' This requires manual identification of extra sheets
+#'
+#' @param ls_df a list of `data.frames` generated by reading in excel sheets
+#' @param remove_sheets `chr` vector of sheet names that should be removed from the list
+#'
+remove_extra_sheets <- function(ls_df, remove_sheets) {
+  # keep elements from `ls_df` that do not match `remove_sheets`
+  ls_df[!(names(ls_df) %in% toupper(remove_sheets))]
+}
+
+#' Select columns to keep from a list of `data.frames`
+#'
+#' Select columns to keep in teh `data.frame` based on partial matching.
+#' Partial matching is required because column names are variable within
+#' each workbook (e.g. "depth" may be any of the following: "Dep100", "Depth (M)", "Dep25", etc)
+#'
+#' @param ls_df a list of `data.frames` generated by reading in excel sheets
+#' @param keep_cols a `chr` vector of column names to keep. Uses partial matching to get around inconsistent naming conventions
+#'
+select_cols_to_keep <- function(ls_df, keep_cols) {
+  keep_cols <- tolower(keep_cols)
+
+  lapply(ls_df, function(x){
+    x %>%
+      # partial match because names are variable - e.g. "Depth", "Depth (M)", "Dep100", etc
+      dplyr::select(contains(keep_cols)) %>%
+      # both secchi depth and depth exist in some data sets
+      dplyr::select(- contains("Secchi"))
+  })
+}
+
+#' Coerce numeric columns
+#'
+#' This function converts user-specified columns to `numeric`. It
+#' assumes that the `Site` column is the only `chr` column.
+#'
+#' @param ls_df a list of `data.frames` generated by reading in excel sheets
+#'
+coerce_numeric_columns <- function(ls_df) {
+  lapply(ls_df, function(df){
+    # ID all the numeric columns
+    num_cols <- c(1:ncol(df)) %>%
+      .[!. %in% grep("site", names(df))] # find and remove site column
+
+    df[num_cols] <- sapply(df[num_cols], as.numeric)
+
+    return(df)
+  })
+}
+
+#' Validate column names
+#'
+#' MO USACE datasets have a wide variety of column names. This function
+#' checks for inconsistencies and fixes them. this function calls two
+#' helper functions: `run_name_checks` and `find_col_idx`
+#'
+#' @param ls_df a list of `data.frames` generated by reading in excel sheets
+#'
+validate_col_names <- function(ls_df) {
```

Comment:
this is a fairly generic function name and maybe only applies to a couple of spreadsheets? If so, you might want to add a word to the function name, since `validate_col_names` seems like a useful concept and maybe it would pop up elsewhere and cause a function name clash.

Reply:
Agreed, I will make this function name a bit more specific. Validating column names is a key piece of getting this data set to munge properly and is needed in almost every year of data provided.